dapr/clients/grpc/client.py

```diff
@@ -451,6 +455,59 @@ def save_state(
         return DaprResponse(
             headers=call.initial_metadata())
 
+    def execute_transaction(
+            self,
+            store_name: str,
+            operations: Sequence[TransactionalStateOperation],
+            metadata: Optional[MetadataTuple] = ()) -> DaprResponse:
+        """Saves key-value pairs to a statestore as a transaction
+
+        This saves key-values to the statestore as part of a single transaction
+        Options for request can be passed with the options field and custom
+        metadata can be passed with metadata field.
+
+        The example saves states to a statestore:
+            from dapr import DaprClient
+            with DaprClient() as d:
+                resp = d.save_state(
+                    store_name='state_store'
+                    states=[{'key': 'key1', 'value': 'value1'}],
+                    etag='etag',
+                    metadata=(
+                        ('header1', 'value1')
+                    ),
+                )
+
+        Args:
+            store_name (str): the state store name to save to
+            key (str): the key to be saved
+            value (bytes or str): the value to be saved
+            etag (str, optional): the etag to save with
+            options (StateOptions, optional): custom options
+                for concurrency and consistency
+            metadata (tuple, optional): custom metadata
+
+        Returns:
+            :class:`DaprResponse` gRPC metadata returned from callee
+        """
+        if not store_name or len(store_name) == 0 or len(store_name.strip()) == 0:
+            raise ValueError("State store name cannot be empty")
+        req_ops = [api_v1.TransactionalStateOperation(
+            operationType=o.operation_type.value,
+            request=common_v1.StateItem(
+                key=o.key,
+                value=to_bytes(o.data),
+                etag=o.etag,
+                metadata=o.metadata)) for o in operations]
+
+        req = api_v1.ExecuteStateTransactionRequest(
+            storeName=store_name,
+            operations=req_ops,
+            metadata=metadata)
```

Comment:
is the same metadata passed in both the gRPC call and here? @youngbupark ?

Reply:
Good point, what do we do with the metadata from proto and the one in the call? Reuse or keep them separate.

Reply:
We need set metadata in `TransactionalStateOperation` message. As mentioned above, this metadata is `transactional_metadata` which is List type, not Tuple.

Reply:
Using separate metadata argument for grpc vs request.